packages/cli/src/environments.ee/source-control/source-control-scoped.service.ts

```diff
@@ -20,6 +24,21 @@ export class SourceControlScopedService {
 		private readonly workflowRepository: WorkflowRepository,
 	) {}
 
+	async ensureIsAllowedToPush(req: AuthenticatedRequest) {
+		if (hasGlobalScope(req.user, 'sourceControl:push')) {
+			return;
+		}
+
+		const ctx = new SourceControlContext(req.user);
+		const projectsWithAdminAccess = await this.getAdminProjectsFromContext(ctx);
+
+		if (Array.isArray(projectsWithAdminAccess) && projectsWithAdminAccess.length > 0) {
```

Comment:
How about :
```ts
if (projectsWithAdminAccess?.length === 0) {
    throw new ForbiddenError('You are not allowed to push changes');
}
```

I think it makes it clearer why the user is forbidden (has no projects with admin access)